samples/web/content/apprtc/js/roomselection.js

```diff
@@ -0,0 +1,139 @@
+  this.roomJoinButton_.addEventListener('click', this.onJoinButton_.bind(this), false);
+  
+  // Public callbacks. Keep it sorted.
+  this.onRoomSelected = null;
+  
+  this.recentlyUsedList_ = new RoomSelection.RecentlyUsedList(recentRoomsKey);
+  this.startBuildingRecentRoomList_();
+};
+
+RoomSelection.prototype.startBuildingRecentRoomList_ = function() {
+  this.recentlyUsedList_.getRecentRooms().then(this.buildRecentRoomList_.bind(this)).catch(function(error) {
+    trace('Error building recent rooms list: ' + error.message);
+  }.bind(this));
+};
+
+RoomSelection.prototype.buildRecentRoomList_ = function(recentRooms) {
+  for (var i = 0; i < recentRooms.length; ++i) {
+    // Create link in recent list
+    var li = document.createElement('li');
+    var href = document.createElement('a');
+    var linkText = document.createTextNode(recentRooms[i]);
+    href.appendChild(linkText);
+    href.href = location.origin + '/r/' + encodeURIComponent(recentRooms[i]);
+    li.appendChild(href);
+    this.roomRecentList_.appendChild(li);
+  
+    // Set up click handler to avoid browser navigation.
+    href.addEventListener('click', this.makeRecentlyUsedClickHandler_(recentRooms[i]).bind(this), false);
+  }
+};
+
+RoomSelection.prototype.onRoomIdInput_ = function() {
+  // validate room id, enable/disable join button
+  // The server currently accepts only the \w character class
+  var room = this.roomIdInput_.value;
+  var valid = room.length >= 5;
+  var re = /^\w+$/;
+  valid = valid && re.exec(room);
+  if (valid) {
+    this.roomJoinButton_.disabled = false;
+  } else {
+    this.roomJoinButton_.disabled = true;
+  }
+};
+
+RoomSelection.prototype.onRandomButton_ = function() {
+  this.roomIdInput_.value = randomString(9);
+};
+
+RoomSelection.prototype.onJoinButton_ = function() {
+  this.loadRoom_(this.roomIdInput_.value);
+};
+
+RoomSelection.prototype.makeRecentlyUsedClickHandler_ = function(roomName) {
+  return function(e) {
+    e.preventDefault();
+    this.loadRoom_(roomName);
+  };
+};
+
+RoomSelection.prototype.loadRoom_ = function(roomName) {
+  this.recentlyUsedList_.pushRecentRoom(roomName);
+  if (this.onRoomSelected) {
+    this.onRoomSelected(roomName);
+  }
+};
+
+RoomSelection.RecentlyUsedList = function(key) {
+  this.RECENTROOMSKEY_ = key || 'recentRooms';
+  this.storage_ = new Storage();
+};
+
+// Add a room to the recently used list and store to local storage.
+RoomSelection.RecentlyUsedList.prototype.pushRecentRoom = function(roomId) {
+  // Push recent room to top of recent list, keep max of 10 entries.
+  return new Promise(function(resolve, reject) {
+    if (!roomId) {
+      resolve();
+      return;
+    }
+    
+    this.getRecentRooms().then(function(recentRooms) {
+      recentRooms = [roomId].concat(recentRooms);
+      // Remove any duplicates from the list, leaving the first occurance.
+      recentRooms = recentRooms.filter(function(value, index, self) {
+        return self.indexOf(value) === index;
```

Comment:
That would work too, the only advantage of this way is that it removes other duplicates as well if they happened to be in the list.